pandas/core/strings.py

```diff
@@ -241,6 +252,117 @@ def g(x):
         return lib.map_infer(arr, f)
 
 
+def str_format(
+    arr,
+    format: str,
+    name: Optional[str] = None,
+    positional_only: bool = False,
+    how_na: str = "any",
+) -> "Series":
+    """
+    Format rows according to the format and return a Series with one string per row.
+
+    Parameters
+    ----------
+    arr : DataFrame or Series
+        The values to format.
+    format : str
+        format string.
+    name : Label, optional
+        The name of the returned Series.
+    positional_only : bool, default False
+        If True, only allow positional parameters (i.e. allow "{}", but not "{key}").
+        Setting to ``True`` will improve performance.
+    how_na : str, one of {"all", "any"}, default "any"
+        If "all", return ``NA`` if all values in row are nan values.
+        If "any", return ``NA`` if at least one of the values in row is a nan value.
+
+    Returns
+    -------
+    Series
+        A Series with dtype ``StringDtype``, formatted according to ``format``.
+
+    Examples
+    --------
+    >>> df = pd.DataFrame({
+    ...     'state_name': ['California', 'Texas', 'Florida'],
+    ...     'state_abbreviation': ['CA', 'TX', 'FL'],
+    ...     'population': [39_512_223, 28_995_881, 21_477_737],
+    ...     }, index=[1, 2, 3])
+    >>> df
+       state_name state_abbreviation  population
+    1  California                 CA    39512223
+    2       Texas                 TX    28995881
+    3     Florida                 FL    21477737
+    >>> ser = df["population"]
+
+    Formatting using positional arguments:
+
+    >>> ser.format("Population: {:,}")
+    1    Population: 39,512,223
+    2    Population: 28,995,881
+    3    Population: 21,477,737
+    dtype: string
+
+    >>> df.format("{} ({}): {:,}")
```

Comment:
Just thinking out loud but would it not be more useful to apply this formatting to the columns returning a like-shaped data frame instead of a series?

Reply:
My thought was that users would rarely want toconvert the whole frame, but usually selected columns and do it like this ``df.assign(x=df['x'].format(...), y=df['y'].format(...))``. Maybe have a ``combine=True`` keyword argument, that if set False, would work on single columns?